lasagne/layers/autoenc.py

```diff
@@ -0,0 +1,262 @@
+
+    Examples
+    --------
+    >>> import lasagne
+    >>> from lasagne.layers import InputLayer, DenseLayer
+    >>> from lasagne.layers import build_autoencoder
+    >>> l_in = InputLayer((100, 20))
+    >>> l1 = DenseLayer(l_in, num_units=50)
+    >>> l2 = DenseLayer(l1, num_units=10)
+    >>> l_ae = build_autoencoder(l2, nonlinearity='same', b=None)
+    """
+
+    if isinstance(nonlinearity, (tuple, list)):
+        n_idx = 0
+
+    if isinstance(b, (tuple, list)):
+        b_idx = 0
+
+    layer = unfold_bias_and_nonlinearity_layers(layer)
+    layers = get_all_layers(layer)
+    autoencoder_layers = [layer]
+
+    for layer in layers[::-1]:
+        if not isinstance(layer, InputLayer):
+            incoming = autoencoder_layers[-1]
+
+            if isinstance(layer, NonlinearityLayer):
+                if nonlinearity == 'same':
+                    kwargs = dict(nonlinearity=layer.nonlinearity)
+                elif nonlinearity is None:
+                    kwargs = dict(nonlinearity=nonlinearities.identity)
+                elif isinstance(nonlinearity, (tuple, list)):
+                    kwargs = dict(nonlinearity=nonlinearity[n_idx])
+                    n_idx += 1
+                else:
+                    kwargs = dict(nonlinearity=nonlinearity)
+                a_layer = NonlinearityLayer(
+                    incoming=incoming,
+                    **kwargs
+                )
+            elif isinstance(layer, DropoutLayer):
+                a_layer = DropoutLayer(
+                    incoming=incoming,
+                    p=layer.p,
+                    rescale=layer.rescale
+                )
+            elif isinstance(layer, GaussianNoiseLayer):
+                a_layer = GaussianNoiseLayer(
+                    incoming=incoming,
+                    sigma=layer.sigma
+                )
+            elif isinstance(layer, BiasLayer):
+                if b is None:
+                    kwargs = dict(b=None)
+                elif b == 'new':
+                    kwargs = dict(b=init.Constant(0.))
+                elif isinstance(b, (tuple, list)):
+                    kwargs = dict(b=b[b_idx])
+                    b_idx += 1
+                else:
+                    kwargs = dict(b=b)
+
+                a_layer = BiasLayer(
+                    incoming=incoming,
+                    **kwargs
+                )
+            else:
+                a_layer = InverseLayer(
+                    incoming=incoming,
+                    layer=layer
+                )
+            autoencoder_layers.append(a_layer)
+
+    return autoencoder_layers[-1]
+
+
+def unfold_bias_and_nonlinearity_layers(layer):
+    """
+    Given a :class:`Layer` instance representing a stacked network,
+    this function adds a :class:`NonlinearityLayer` instance in between
+    each layer with attribute nonlinearity, with the same nonlinearity,
+    while setting the nonlinearity of the original layer to the identity
+    function.
```

Comment:
Again, that's too long for a first sentence, and too short for a second sentence because it misses the explanation for `BiasLayer`.